Source/WTF/wtf/SHA1.cpp

```diff
@@ -203,6 +204,27 @@ void SHA1::reset()
 
 #endif
 
+void SHA1::addUTF8Bytes(StringView string)
+{
+    if (string.containsOnlyASCII()) {
+        if (string.is8Bit())
+            addBytes(string.span8());
+        else
+            addBytes(String::make8Bit(string.span16()).span8());
+    } else
+        addBytes(string.utf8().span());
+}
+
+#if USE(CF)
+void SHA1::addUTF8Bytes(CFStringRef string)
+{
+    if (auto* characters = CFStringGetCStringPtr(string, kCFStringEncodingASCII))
+        addBytes(std::span { byteCast<uint8_t>(characters), static_cast<size_t>(CFStringGetLength(string)) });
+    else
+        addUTF8Bytes(String(string));
```

Comment:
Can write even more code to avoid allocating a `String` when the `CFStringRef` is all ASCII but `CFStringGetCStringPtr` returns `nullptr`. It would use `CFStringGetBytes`.

Reply:
I'll look into that.

Reply:
I'm thinking of using `CFStringGetCString` which seems simpler to use and is documented as fall back to `CFStringGetCStringPtr`.

Reply:
```
void SHA1::addUTF8Bytes(CFStringRef string)
{
    if (auto* characters = CFStringGetCStringPtr(string, kCFStringEncodingASCII)) {
        addBytes(std::span { byteCast<uint8_t>(characters), static_cast<size_t>(CFStringGetLength(string)) });
        return;
    }

    CFIndex maxLength = CFStringGetMaximumSizeForEncoding(CFStringGetLength(string), kCFStringEncodingASCII);
    Vector<char, 1024> buffer(maxLength);
    if (CFStringGetCString(string, buffer.data(), buffer.size(), kCFStringEncodingASCII)) {
        addBytes(span8(buffer.data()));
        return;
    }

    addUTF8Bytes(String(string));
}
```